### DISCUSSION TITLE
Does using an override cause vcpkg to ignore the features array?

### DISCUSSION THREAD

**Discussion by cmannett85** [microsoft/vcpkg#26654] [General]
Let's say I have a `vcpkg.json` that looks like this:
```
{
    "name": "foo",
    "version-string": "3.14",
    "builtin-baseline": "cef0b3ec767df6e83806899fe9525f6cf8d7bc91",
    "dependencies": [
        {
            "name": "boost-asio",
            "features": []
        }
    ],
    "overrides": [
        {
            "name": "boost-asio",
            "version": "1.74.0"
        }
    ]
}
```
The latest `boost-asio` for the given baseline is v1.77, but as I have provided an override, vcpkg duly drops the built version to v1.74.  _However_ when I do this for a real project I find that vcpkg also builds openssl even though the feature has been disabled:
```
  * boost-winapi[core]:x64-linux -> 1.74.0 -- /home/cmannett85/workspace/arg_router/vcpkg/buildtrees/versioning_/versions/boost-winapi/62117b89dedc1807c7de0bfce139141aedeb8bd7
  * openssl[core]:x64-linux -> 3.0.3#2 -- /home/cmannett85/workspace/arg_router/vcpkg/buildtrees/versioning_/versions/openssl/cdd8b7c5c9ebba05598bc90a58f224f13cf0b611
```
It looks a lot like using overrides sets the features list back to the package defaults - is this a bug, or am I doing it wrong?

P.S.  Before anyone says I should just be [modifying the baseline instead](https://vcpkg.io/en/docs/examples/modify-baseline-to-pin-old-boost.html#method-2-modify-baseline), I use vcpkg as a submodule and shouldn't be expected to fork the repo so others can access my modified version, just for versioning to work...

**Comment by ras0219-msft** [comment]
`boost-asio` in version 1.74.0 has a hard dependency on openssl[1], it was moved into a feature in version 1.78.0 (https://github.com/microsoft/vcpkg/pull/21928).

If you want 1.74.0 without openssl, the easiest way is create your own overlay port from the 1.74.0 version that doesn't depend on openssl.

[1]: https://github.com/microsoft/vcpkg/blob/0e1252388d5353f031e177b2826bab98d6c2e21c/ports/boost-asio/CONTROL#L4

**Reply by cmannett85** [reply]
Ah thank you!  I didn't even bother checking the vcpkg.json from the version I was complaining about - shame on me...
